virtual_rainforest/core/config.py

```diff
@@ -298,12 +298,16 @@ class Config(dict):
     Args:
         cfg_paths: A string, Path or list of strings or Paths giving configuration
             file or directory paths.
+        extra_params: Extra parameters provided by the user.
         auto: flag to turn off automatic validation.
 
     """
 
     def __init__(
-        self, cfg_paths: Union[str, Path, list[Union[str, Path]]], auto: bool = True
+        self,
+        cfg_paths: Union[str, Path, list[Union[str, Path]]],
+        extra_params: Optional[list[dict[str, Any]]] = None,
```

Comment:
I'm curious as to why this uses `Optional` and `None` when the custom attribute is immediately  set to be an empty list if it isn't provided. As opposed to doing:

```suggestion
        extra_params: list[dict[str, Any]] = [],
```

That is then a mutable default, which is dangerous, but it could be a tuple of dicts instead.

Reply:
Good point. We're `deepcopy`ing it anyway, so it doesn't matter if it's mutable.